drivers/SmartThings/matter-sensor/src/init.lua

```diff
@@ -175,15 +303,26 @@ local function do_configure(driver, device)
   device:try_update_metadata({profile = profile_name})
 end
 
+local function do_configure(driver, device)
+  match_profile(driver, device)
+end
+
 local function device_init(driver, device)
   log.info("device init")
   device:subscribe()
 end
 
 local function info_changed(driver, device, event, args)
   if device.profile.id ~= args.old_st_store.profile.id then
-    device:subscribe()
     set_boolean_device_type_per_endpoint(driver, device)
+    for cap_id, attributes in pairs(subscribed_attributes) do
```

Comment:
Similar to my message above, this shouldn't be needed. This same loop is included in the base definition of the `subscribe` function: https://github.ecodesamsung.com/iot-hub/scripting-engine/blob/c401bb4dfece0158aa0903e3843e81e383ff09e7/lua_libs/st/matter/device.lua#L242

Reply:
Updated in latest commit!